pkg/network/setup/netstat.go

```diff
@@ -138,6 +138,18 @@ func (c *NetStat) UpdateStatus(vmi *v1.VirtualMachineInstance, domain *api.Domai
 		}
 	}
 
+	indexedIfaceStatus := netvmispec.IndexInterfacesFromStatus(interfacesStatus, func(ifaceStatus v1.VirtualMachineInstanceNetworkInterface) bool { return true })
```

Comment:
I think you just need to adjust the previous iteration (on line 134).

Instead of iterating over the `interfacesStatus`, there is a need to iterate over the `multusStatusNetworksByName` and perform a lookup for the interface in the status.
If it is there, add the source (as now).
If it is not there, add a new entry as you do in this new block.

Having two iterations is confusing and redundant IMO.